stubs/tensorflow/tensorflow/keras/callbacks.pyi

```diff
@@ -0,0 +1,190 @@
+
+class EarlyStopping(Callback):
+    monitor_op: Any
+    def __init__(
+        self,
+        monitor: str = "val_loss",
+        min_delta: float = 0,
+        patience: int = 0,
+        verbose: Literal[0, 1] = 0,
+        mode: Literal["auto", "min", "max"] = "auto",
+        baseline: float | None = None,
+        restore_best_weights: bool = False,
+        start_from_epoch: int = 0,
+    ) -> None: ...
+
+class History(Callback):
+    history: dict[str, list[Any]]
+
+class LambdaCallback(Callback):
+    def __init__(
+        self,
+        on_epoch_begin: Callable[[int, _Logs], object] | None = None,
+        on_epoch_end: Callable[[int, _Logs], object] | None = None,
+        on_batch_begin: Callable[[int, _Logs], object] | None = None,
+        on_batch_end: Callable[[int, _Logs], object] | None = None,
+        on_train_begin: Callable[[_Logs], object] | None = None,
+        on_train_end: Callable[[_Logs], object] | None = None,
+        **kwargs: Any,
+    ) -> None: ...
+
+class LearningRateScheduler(Callback):
+    def __init__(
+        self,
+        schedule: LearningRateSchedule | Callable[[int], float | tf.Tensor] | Callable[[int, float], float | tf.Tensor],
+        verbose: Literal[0, 1] = 0,
+    ) -> None: ...
+
+class ModelCheckpoint(Callback):
+    monitor_op: Any
+    filepath: str
+    _options: tf.train.CheckpointOptions | tf.saved_model.SaveOptions | None
+    def __init__(
+        self,
+        filepath: str,
+        monitor: str = "val_loss",
+        verbose: Literal[0, 1] = 0,
+        save_best_only: bool = False,
+        save_weights_only: bool = False,
+        mode: Literal["auto", "min", "max"] = "auto",
+        save_freq: str | int = "epoch",
+        options: tf.train.CheckpointOptions | tf.saved_model.SaveOptions | None = None,
+        initial_value_threshold: float | None = None,
+    ) -> None: ...
+    def _save_model(self, epoch: int, batch: int | None, logs: _Logs) -> None: ...
+
+class ProgbarLogger(Callback):
+    use_steps: bool
+    def __init__(
+        self, count_mode: Literal["steps", "samples"] = "samples", stateful_metrics: Iterable[str] | None = None
+    ) -> None: ...
+
+class ReduceLROnPlateau(Callback):
+    def __init__(
+        self,
+        monitor: str = "val_loss",
+        factor: float = 0.1,
+        patience: int = 10,
+        verbose: Literal[0, 1] = 0,
+        mode: Literal["auto", "min", "max"] = "auto",
+        min_delta: float = 1e-4,
+        cooldown: int = 0,
+        min_lr: float = 0,
+        **kwargs: Incomplete,
+    ) -> None: ...
+    def in_cooldown(self) -> bool: ...
+
+class RemoteMonitor(Callback):
+    def __init__(
+        self,
+        root: str = "http://localhost:9000",
+        path: str = "/publish/epoch/end/",
+        field: str = "data",
+        headers: _HeadersMapping | None = None,
+        send_as_json: bool = False,
+    ) -> None: ...
+
+class TensorBoard(Callback):
+    _start_batch: int
```

Comment:
Any particular reason to include these private attributes? We generally aim to add private attributes only if someone explicitly asks for them.

Reply:
These probably come from my stubs and are attributes I've accessed before.

I'd expect most of these are rarely used and are fine just type ignoring them on my end, so neutral to dropping them.

Reply:
Ok, I'll remove the private attributes then. We can always re-add them later if requested.